replica/table.cc

```diff
@@ -2366,7 +2366,7 @@ future<> table::move_sstables_from_staging(std::vector<sstables::shared_sstable>
             // That being said, we'll only add this SSTable to tracker if its origin is other than repair.
             // Otherwise, we can count on off-strategy completion to add it when updating lists.
             if (sst->get_origin() != sstables::repair_origin) {
-                add_sstable_to_backlog_tracker(_compaction_strategy.get_backlog_tracker(), sst);
+                add_sstable_to_backlog_tracker(compaction_group_for_sstable(sst).get_backlog_tracker(), sst);
```

Comment:
That's a good question. Without MV for example, we'll probably want to segregate streaming data across groups, because the idea is that during normal flow (no split / merge of groups), each sstable will be owned by a single group. Alternatively, we could have a special group for staging data like you suggested. Food for thought :-)

Reply:
I think it's a good idea for running cleanup in the staging dir: https://github.com/scylladb/scylladb/issues/9559#issuecomment-1282488329

Reply:
There are several ways to lay this out:

1. separate arrays
 
table -> main compaction groups [idx] -> sstables
table -> staging compaction groups [idx] -> sstables

smells bad

2. one array, complex compaction group

table -> compaction groups [idx] -> compaction_group { sstable_set main, sstable_set staging }

3. one array, of objects holding compaction groups:

table -> compaction_groups [ idx] -> something { compaction_group main, compaction_group staging }

This allows the two sub-groups to have different compaction strategy

Reply:
With off-strategy compaction becoming the default for streaming and repair based ops: https://github.com/scylladb/scylladb/pull/11848

we can probably kill that snippet as staging sstables will be in maintenance set (off-strategy kicks in after we're done with view building only) and won't belong to the backlog tracker.

Reply:
I think repair / streaming will probably have to segregate the incoming data to their owner groups, one sstable per group. This reduces write amplification by 1, as we won't have to read and re-write all the staging data (while performing segregation).